pkg/build/admission/overrides/api/types.go

```diff
@@ -17,4 +19,10 @@ type BuildOverridesConfig struct {
 	// If user provided a label in their Build/BuildConfig with the same name as one in this
 	// list, the user's label will be overwritten.
 	ImageLabels []buildapi.ImageLabel
+
+	// nodeSelector is a selector which must be true for the build pod to fit on a node
+	NodeSelector map[string]string
```

Comment:
my view of the default nodeselector was "must be at least this specific".

there are probably competing use cases.  but my assumption is most admins are going to set their system up with node labels like "buildNode=foo", and a defaulter that assigns the buildNode nodeselector key, which ensures that all builds will land on a node with at least the "buildNode" key, of some value.